be/src/storage/lake/horizontal_compaction_task.cpp

```diff
@@ -49,7 +50,8 @@ Status HorizontalCompactionTask::execute(Progress* progress) {
     reader_params.use_page_cache = false;
     RETURN_IF_ERROR(reader.open(reader_params));
 
-    ASSIGN_OR_RETURN(auto writer, _tablet->new_writer(kHorizontal));
+    int64_t txn_id = next_id();
+    ASSIGN_OR_RETURN(auto writer, _tablet->new_writer(kHorizontal, txn_id))
```

Comment:
should also add txn id in vertical compaction

Reply:
Fixed